test/OrchardCore.Tests/Modules/OrchardCore.Resources/SubResourceIntegrityTests.cs

```diff
@@ -0,0 +1,73 @@
+using System.Security.Cryptography;
+using OrchardCore.ResourceManagement;
+using OrchardCore.Resources;
+
+namespace OrchardCore.Tests.Modules.OrchardCore.Resources
+{
+    public class SubResourceIntegrityTests
+    {
+        [Fact]
+        public void CheckSubResourceIntegrity()
+        {
+            // Arrange
+            var resourceOptions = Options.Create(new ResourceOptions());
+            var httpContextAccessorMock = new Mock<IHttpContextAccessor>();
+            httpContextAccessorMock
+                .Setup(a => a.HttpContext)
+                .Returns(new DefaultHttpContext());
+            var configurationOptions = new ResourceManagementOptionsConfiguration(
+                resourceOptions,
+                Mock.Of<IHostEnvironment>(),
+                httpContextAccessorMock.Object
+            );
+            var resourceManagementOptions = new ResourceManagementOptions();
+
+            // Act & Assert
+            configurationOptions.Configure(resourceManagementOptions);
+
+            var resourceManifest = resourceManagementOptions.ResourceManifests.First();
+
+            ValidateSubResourceIntegrity("script");
+            ValidateSubResourceIntegrity("style");
+
+            void ValidateSubResourceIntegrity(string resourceType)
+            {
+                foreach (var resource in resourceManifest.GetResources(resourceType))
+                {
+                    foreach (var resourceDefinition in resource.Value)
+                    {
+                        if (!String.IsNullOrEmpty(resourceDefinition.CdnIntegrity) && !String.IsNullOrEmpty(resourceDefinition.UrlCdnDebug))
+                        {
+                            var resourceIntegrity = GetSubResourceIntegrityAsync(resourceDefinition.UrlCdnDebug).Result;
+
+                            Assert.True(resourceIntegrity.Equals(resourceDefinition.CdnDebugIntegrity), $"The {resourceType} {resourceDefinition.UrlCdnDebug} has invalid SRI hash, please use '{resourceIntegrity}' instead.");
```

Comment:
Some line breaks here would be nice so it's not that long of a line, like along arguments. Below too.